src/storage/store/csr_node_group.cpp

```diff
@@ -117,18 +157,79 @@ NodeGroupScanResult CSRNodeGroup::scan(Transaction* transaction, TableScanState&
 }
 
 NodeGroupScanResult CSRNodeGroup::scanCommittedPersistent(const Transaction* transaction,
-    const RelTableScanState& tableState, CSRNodeGroupScanState& nodeGroupScanState) const {
-    if (nodeGroupScanState.nextRowToScan == nodeGroupScanState.persistentCSRList.length) {
-        return NODE_GROUP_SCAN_EMMPTY_RESULT;
+    RelTableScanState& tableState, CSRNodeGroupScanState& nodeGroupScanState) const {
+    while (true) {
+        while (nodeGroupScanState.tryScanCachedTupels(tableState)) {
+            if (tableState.outState->getSelVector().getSelSize() > 0) {
+                // Note: This is a dummy return value.
+                return NodeGroupScanResult{nodeGroupScanState.numScannedRows,
+                    tableState.outState->getSelVector().getSelSize()};
+            }
+        }
+        if (nodeGroupScanState.numScannedRows == nodeGroupScanState.numTotalRows ||
```

Comment:
We only cache for scanCommittedPersistent but NOT for CommittedInMem or UnCommitted. I'm assuming this is because there is no noticeable performance improvement if we do so?

Reply:
Yeah, that's my assumption for now.